### ISSUE TITLE
should git objects know whats above them?

### ISSUE THREAD

**Issue by andrewseddon** [circuithub/massive-git#1]
So in your current implementation your planning on Git objects, such as blob's knowing which repo's they are part of? Git doesn't actually do this and i'm wondering if it's the way to go (havent decided).
## Pros
- If we use Riak search to find info in blobs it would be useful to see which repo that blob is part off.
- Garbage collection will be easier. GC without reverse links will be very labour intensive as we'll have to check if a blob is referenced anywhere before deleting it.
## Cons
- Extra logic to maintain these links
- Possible issue if a blob becomes part of lots of repo's. Say the repo get's forked by 500 users each blob will now need a reference to each of those.

**Comment by podviaznikov** [comment]
Thanks for the comments.  Please look into the code as much as you can as ask questions or make propositions. I still don't have full picture in my mind.

I took a lot of ideas from grit(https://github.com/mojombo/grit/blob/master/lib/grit/git-ruby/git_object.rb) and nodejs implementation for Git. In these projects they had optional link to repository from each GitObject.

I will think more about this.

**Comment by andrewseddon** [comment]
Yeah definitely follow grit and https://github.com/christkv/node-git . Optional link sounds like a good idea.
